### PR TITLE
Refactor accordion component

### LINKED ISSUE #12676: Update Accordion components for extensibility and flexibility
The Accordion component was created for 0.17 to allow for better management of space and to also allow user control over seeing more and less details. 

For 0.18, we will continue to use an Accordion pattern, and while the overall use is the same (a toggle open and close), the use cases are more varied. There is also some amount of general cleanup of the Accordion that can be done in the course of this extension. This should _not_ be a comprehensive refactor, but rather the focus should be ensuring that the uses of the accordion are supported, that any small updates that make the accordion easier to use are implemented, and that we set a strong foundation for the API as we have been discussing moving the accordion to KDS. 

The following places are where the accordion is used in 0.18
| Description | Screenshot |
|---|---
| The new "selected resources" preview - a "shopping cart" feature that allows the coach to see which items they've currently selected to add to the lesson or quiz, while navigating through searches or up and down the topic tree  |  Screenshot 2024-09-24 at 10 09 17 AM |
| In quizzes - when previewing the quiz (new) or when looking at the quiz as a learner, or the quiz report as a coach. The accordions allow for separating the quiz sections  |  Screenshot 2024-09-24 at 10 11 08 AM |
| In the search filters panel in coach, to manage filter options (note that this is a variation of the current implementation of the search filters panel)  |  Screenshot 2024-09-24 at 10 09 55 AM |
| Question previewing in the quiz editing side panel  |  Screenshot 2024-09-24 at 10 09 39 AM |
| When used for example in the "filter" panel, the accordion reflects the selection state of it's child items | Screenshot 2024-09-25 at 11 40 56 AM |
| Alternatively, if no options within the accordion may be selected, the accordion is inactive | Screenshot 2024-09-25 at 11 41 16 AM |

### Tasks

- Accordion components and related composables are updated as needed to support all interactions outlined above, including potentially multiple different types of interactions (checkboxes, "accordion item is a button", and icon buttons as part of the accordion item slot)

### Acceptance Criteria 
- Accordion contains a flexible slot that supports both text display as well as interactive elements within the accordion that could be one or more semantic elements (button, li, p, etc.)
- Accordion header can reflect a selected state that matches the selection of one or more of its children
- Accordion header can be inactive if no matching options exist at the current "filter" level ( _note_: please speak to Radina about accessibility implications. loop in Marcella and Jessica as needed.)
- documentation (code comments) is updated accordingly

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Summary

Refactor `AccordionContainer` and `AccordionItem` components to have a more flexible and reusable API.

For this:
* (temporarily) I have duplicated the current `AccordionContainer`, `AccordionItem` components, and `useAccordion` composable into a `accordion` folder: https://github.com/learningequality/kolibri/pull/12693/files#diff-97a18f3c66e485eb7d161548e08b67828e09217ace1a8ea8cde36c615bed8fd6. And have updated these files instead of breaking the current ones.
* (temporarily) I have created a playground page: http://localhost:8000/es-es/coach/#/playground to test different styles of accordions.
* I have updated the current most complex implementation of Accordions to use the new components with the draggable questions in [CreateQuizSection](https://github.com/AlexVelezLl/kolibri/blob/522196994aaf7a80cbe4ca57d789f4ed06eaf8e7/kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/CreateQuizSection.vue#L171).

Changes made to the `AccordionContainer`, `AccordionItem` components, and `useAccordion` composable:
* The main change is that the user wont need to know about the `useAccordion` composable, as this have been changed for internal use of `AccordionContainer` and `AccordionItem`, This by using the `provide/inject` pattern:
  * `AccordionContainer` will call the `useAccordion` composable, and [this will provide](https://github.com/learningequality/kolibri/pull/12693/files#diff-d15f81d75120918472935d81983765d40da3c02f6139ef2965c85f2c8852b76eR52) the `registerItem`, `unregisterItem`, `toggle` and `isExpanded` functions. Then `AccordionItem`'s will need to inject these properties using the [injectAccordionItem function](https://github.com/learningequality/kolibri/pull/12693/files#diff-d15f81d75120918472935d81983765d40da3c02f6139ef2965c85f2c8852b76eR65).
  *  For `AccordionContainer` to know what items it has defined inside (useful for the `expandAll` behaviour), every `AccordionItem` will be registered on mounted to be part of the accordion items of its closer AccordionContainer parent.
  * Note that this works even with nested accordions (as you can see in the screencast section) because the `inject` method will only inject the properties that were provied by a closer ancestor component in the component hierarchy.
  * The `canExpandAll`, `expandAll`, `canCollapseAll`, `collapseAll` functions will be exposed to the user as slot props of the #header slot of the `AccordionContainer` component: https://github.com/learningequality/kolibri/pull/12693/files#diff-1c7f14aac16f550d4cf8838ee32f3629ed546cc9dd10bf5574a1e6106ff718ceR62. This makes that the user dont need to call the `useAccordion` composable, having less boilerplate code.
* AccordionContainer #header slot will have some pre-defined styles that are the most common across the examples, but the AccordionContainer component will have an `headerAppearanceOverrides` to allow the consumer to override these header wrapper element styles.
* The current AccordionItem component has almost no responsibility. With this new version this component will have more responsibilities to toggle the expaned state, and to ensure best accessibility practices (to be discussed and improved) https://github.com/learningequality/kolibri/pull/12693/files#diff-7c4ec6c2ffd87d923f31a797c4e6e6522a1af8912ef7cf828cd1bac93aeea16fR12.
* For now the Accordion item component exposes a little more rigid structure, having a `title` prop and `leading-actions` and `trailing-action` slots. So we dont need to re-write the common styles of the item header, and this satisfies and reuses the logic for all current examples of accordions. If in the future we need something more flexible we can add a #header slot that has total control on what its being rendered as item header.
* The accordion item component also provides the `headerAppearanceOverrides` and `contentAppearanceOverrides` props to fully customize header and content wrapper elements styles.
* Another change compared to the current implementation of accordions is that now when we navigate with keyboard, we first focus the whole acoordion item header and after that we select anything inside (leading or trailing actions). To have a similar behaviour to how KCard works.

### Screencast

https://github.com/user-attachments/assets/f06da4f1-0d06-4be4-85a7-80b764c76a27

## References
Closes #12676.

## Reviewer guidance
Play with the accordions of the CreateQuizSection, and with the components available in the playground page.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: packages/kolibri-common/components/accordion/AccordionContainer.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  import { computed } from 'kolibri.lib.vueCompositionApi';
+  import useAccordion from './useAccordion';
+
+  export default {
+    name: 'AccordionContainer',
+    setup(prop, { slots }) {
+      const { canExpandAll, canCollapseAll, collapseAll, expandAll } = useAccordion();
+
+      const hasHeaderSlot = computed(() => !!slots.header);
```
Comment (AlexVelezLl) [inline packages/kolibri-common/components/accordion/AccordionContainer.vue:39]: For some reason it seems there is a bug when we use slots with slot-props and the vue Composition API, which is that outside the `setup` method, `this.$slots` is usually empty even if we do have slots.

File: kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/CreateQuizSection.vue [hunk 2]
```diff
@@ -168,43 +168,48 @@
...
-            <KIconButton
-              icon="trash"
-              :tooltip="coreString('deleteAction')"
-              :aria-label="coreString('deleteAction')"
-              :disabled="selectedActiveQuestions.length === 0"
-              @click="() => deleteQuestions()"
-            />
+        <AccordionContainer>
+          <template #header="{ canExpandAll, expandAll, canCollapseAll, collapseAll }">
+            <div style="display: flex; align-items: center; justify-content: space-between">
```
Comment (AlexVelezLl) [inline kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/CreateQuizSection.vue]: All these inline styles should be updated in the future, for now I have used them to quick prototype how the new api would look like

**Review by nucleogenesis — COMMENTED** [review]
@AlexVelezLl this is awesome :) 

I really like the use of leading/trailing-actions for the slots for things like checkboxes and context menus on the edges of the header.

Everything looks great to me and like it'll be pretty easy and straightforward to use for the many places we use this. I'd love to follow-up and update uses of the Accordion in a few places to get a real feel for the differences.

I won't give approval until you're ready for a fuller review, but I have no changes to request aside from removing the playground and such before merging.

**Review by nucleogenesis — APPROVED** [review]
Gave this another look and I think this should be good to merge so long as we have some follow-up issues to replace the old Accordion w/ the new one. But I think we'll look to merge this sooner-than-later so that we can get started using the updated Accordion as we move forward on coach resource discovery updates.

**Pull request merged**